charts/opentelemetry-operator/templates/deployment.yaml

```diff
@@ -49,6 +49,8 @@ spec:
             {{- end }}
             {{- if and .Values.manager.collectorImage.repository .Values.manager.collectorImage.tag }}
             - --collector-image={{ .Values.manager.collectorImage.repository }}:{{ .Values.manager.collectorImage.tag }}
+            {{- else if .Values.manager.collectorImage.repository }}
+            - --collector-image={{ .Values.manager.collectorImage.repository }}
```

Comment:
Let's move this and the above logic into a helper template so we don't have to do the same condition twice.

Reply:
Do you mean, something like this?
```
{{- define "opentelemetry-operator.collectorImage" -}}
  {{- if and .Values.manager.collectorImage.repository .Values.manager.collectorImage.tag }}
    - --collector-image={{ .Values.manager.collectorImage.repository }}:{{ .Values.manager.collectorImage.tag }}
  {{- else if .Values.manager.collectorImage.repository }}
    - --collector-image={{ .Values.manager.collectorImage.repository }}
  {{- end -}}
{{- end }}
```

Reply:
I believe we can do something like:

```
{{ - $tag = "" }}
{{ if .Values.manager.collectorImage.tag }}
{{ - $tag = ":" + .Values.manager.collectorImage.tag }}
{{- end }}
- --collector-image={{ .Values.manager.collectorImage.repository }}{{ $tag }}
```

Reply:
BTW, despite you are forcing the users to add a collector repository. I would add an if condition here so just in case you remove the requirement in the future, it can avoid a bug in case you forget to add the if condition here later.
Check it out:

```
{{- if .Values.manager.collectorImage.repository -}}
{{- $collectorImageTag := "" -}}
{{- if .Values.manager.collectorImage.tag -}}
{{- $collectorImageTag = print ":" .Values.manager.collectorImage.tag -}}
{{- end }}
- --collector-image={{ .Values.manager.collectorImage.repository }}{{ $collectorImageTag }}
{{- end }}
```